src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -11,113 +11,342 @@
+            throw new TimeoutException();
 
-	static class Extensions
-	{
-		private const uint Wgs84SpatialReferenceId = 4326;
-
-		public static Geoposition ToGeoPosition(this Location location)
-			=> new Geoposition(
-				new Geocoordinate(
-					latitude: location.Latitude,
-					longitude: location.Longitude,
-					altitude: location.Altitude,
-					timestamp: FromUnixTime(location.Time),
-					speed: location.HasSpeed ? location.Speed : 0,
-					point: new Geopoint(
-						new BasicGeoposition
-						{
-							Latitude = location.Latitude,
-							Longitude = location.Longitude,
-							Altitude = location.Altitude,
-						},
-						AltitudeReferenceSystem.Ellipsoid,
-						Wgs84SpatialReferenceId
-					),
-					accuracy: 0,
-					altitudeAccuracy: 0,
-					heading: null
-				)
-			);
-
-		private static DateTimeOffset FromUnixTime(long time)
-		{
-			var epoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
-			return epoch.AddMilliseconds(time);
-		}
-	}
+        }
+
+        public static async Task<Windows.Devices.Geolocation.GeolocationAccessStatus> RequestAccessAsync()
+        {
+
+            // below 6.0 (API 23), permission are granted
+            if (Android.OS.Build.VERSION.SdkInt < Android.OS.BuildVersionCodes.M)
+            {
+                return Windows.Devices.Geolocation.GeolocationAccessStatus.Allowed;
+            }
+
+            // do we have declared this permission in Manifest?
+            // it could be also Coarse, without GPS
+            Android.Content.Context context = Android.App.Application.Context;
+            Android.Content.PM.PackageInfo packageInfo = 
+                context.PackageManager.GetPackageInfo(context.PackageName, PackageInfoFlags.Permissions);
+            var requestedPermissions = packageInfo?.RequestedPermissions;
+            if(requestedPermissions is null)
+                return Windows.Devices.Geolocation.GeolocationAccessStatus.Denied;
+            
+            bool bInManifest = false;
+            foreach (string oPerm in requestedPermissions)
+            {
+                if (oPerm.Equals(Android.Manifest.Permission.AccessFineLocation, StringComparison.OrdinalIgnoreCase))
+                    bInManifest = true;
+
+            }
+
+            if(!bInManifest)
+                return Windows.Devices.Geolocation.GeolocationAccessStatus.Denied;
+
+            // check if permission is granted
+            if (Android.Support.V4.Content.ContextCompat.CheckSelfPermission(Uno.UI.ContextHelper.Current, Android.Manifest.Permission.AccessFineLocation)
+                    == Android.Content.PM.Permission.Granted)
+            {
+                return Windows.Devices.Geolocation.GeolocationAccessStatus.Allowed;
+            }
+
+            // now, system dialog asking for permission should be displayed - see PR2024
+			
+            return Windows.Devices.Geolocation.GeolocationAccessStatus.Denied;
+            
+        }
+
+
+        static bool mbZmiany = false;
```

Comment:
`mbZmiany` ? Please be more descriptive.

Reply:
also, from polish - "zmiany" means "changes" :)
converted to _locChanged